### ISSUE TITLE
NPM Package excludes source and library features.

### ISSUE THREAD

**Issue by pbower** [ghiscoding/slickgrid-react#233]
### Describe the bug

.npmignore contains the src package, meaning the package is not runnable following .npm install.

The only 5 files that appear to download with the .npm install are:

global.d.ts
LICENSE
package.json
tsconfig.json
README.md

Thanks!

### Reproduction

npm install slick-grid-react

### Expectation

npm install slick-grid-react install results in being able to import and run the core SlickGrid-React modules in Typescript/React.

### Environment Info

```shell
Bash Ubuntu 22.04.3 LTS
Typescript
VsCode
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/ghiscoding/slickgrid-react/blob/master/.github/CODE_OF_CONDUCT.md)
- [X] Read the [HOWTO - Step by Step](https://github.com/ghiscoding/slickgrid-react/wiki/HOWTO--Step-by-Step).
- [X] Check that there isn't [already an issue](https://github.com/ghiscoding/slickgrid-react/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/ghiscoding/slickgrid-react/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by ghiscoding** [comment]
Sorry but I don't really understand your question, why would you want the source code? What do you mean by runnable? All you typically ever need is the distribution files, would you need the source files? You would need to provide mode details and perhaps code sample because I don't understand what the problem is here. 

Also note that I'm not a React developer, someone else created the project (from my other Slickgrid projects like Angular-Slickgrid), so perhaps it's a React thing that I don't understand.

**Comment by pbower** [comment]
That makes sense - my apologies, there were some issues on my end, meaning the dist folder wasn't showing.
In the current version, a heads up these modules are not resolvable and cause errors, although I believe this is likely a config resolution with webpack / craco on my end, relating to the parent universal module resolving from slickgrid-react. I am including here in case it is worth including any additional setup instructions in https://github.com/ghiscoding/slickgrid-react/wiki/HOWTO--Step-by-Step .

ERROR in ./node_modules/slickgrid-react/dist/esm/components/slickgrid-react.js 8:0-30
Module not found: Error: Can't resolve 'slickgrid/slick.core' in '<project>/node_modules/slickgrid-react/dist/esm/components'

ERROR in ./node_modules/slickgrid-react/dist/esm/components/slickgrid-react.js 9:0-38
Module not found: Error: Can't resolve 'slickgrid/slick.interactions' in '<project>/node_modules/slickgrid-react/dist/esm/components'

ERROR in ./node_modules/slickgrid-react/dist/esm/components/slickgrid-react.js 10:0-30
Module not found: Error: Can't resolve 'slickgrid/slick.grid' in '<project>/node_modules/slickgrid-react/dist/esm/components'

ERROR in ./node_modules/slickgrid-react/dist/esm/components/slickgrid-react.js 11:0-34
Module not found: Error: Can't resolve 'slickgrid/slick.dataview' in '<project>/node_modules/slickgrid-react/dist/esm/components'

As a workaround, I replaced these imports: 
import 'slickgrid/slick.core'; 
import 'slickgrid/slick.interactions';
import 'slickgrid/slick.grid';
import 'slickgrid/slick.dataview';

With '../../../../../node_modules/slickgrid/dist/esm/index'; 

I also tried webpack.config.js and craco.config.js aliasing for Slickgrid, but encountered issues.

Assuming this is config-related please feel free to close this, however if there are any recommended setup steps here for the 'How to guide' would be greatly appreciated.

Regards,
Peter

**Comment by ghiscoding** [comment]
This is unlikely a problem with the library itself since the demo works properly, I always check that it works before & after publishing a new release. I don't need to modify any of the imports, it's likely a problem in your configuration, you should really compare against the [Slick-React-Demos](https://github.com/ghiscoding/slickgrid-react-demos), it is using WebPack and it works without any problem.

> Package.json:
> "slickgrid": "latest",
> "slickgrid-react": "latest",
> "slickgrid-universal": "github:ghiscoding/slickgrid-universal",

Well this is likely 1 of the major problem you have, the latest version of SlickGrid (5.0) is **not compatible** with my other libs and I'm not sure why you're overriding it to use the latest since I know it will fail (the dist files are not located at the same place and the build is totally different). You should never override the SlickGrid core library, you're looking for trouble when doing that. So don't do that because it will fail. I am working on that next version as a major and breaking change and that is likely a couple months away (for example this Slickgrid-Universal [PR](https://github.com/ghiscoding/slickgrid-universal/pull/1135) is towards that goal but there is still a lot of work to do for the next major). 

So in summary, I strongly advise you to look at the [Slick-React-Demos](https://github.com/ghiscoding/slickgrid-react-demos) and if the HOWTO is missing something, it could be updated, it was done a while ago so some instructions might be out of sync or incorrect.
